ops/terraform/modules/resources/bfd_pipeline/main.tf

```diff
@@ -134,6 +134,41 @@ resource "aws_security_group_rule" "allow_db_primary_access" {
   source_security_group_id = aws_security_group.app.id # The EC2 instance for the BFD Pipeline app.
 }
 
+# NACL to manage communication between the BFD and RDA environments.
+# The NACL rules will open communication over the known ports and close
+# everything else.  This is done by creating a "lower" rule that explicitly
+# allows communication and defining a "upper" rule that defaults to denying
+resource "aws_network_acl" "rda" {
+  # only create the NACL if the CIDR block has been defined
+  count = var.mpm_rda_cidr_block != null ? 1 : 0
+
+  vpc_id      = var.env_config.vpc_id
+  tags        = merge({ Name = "bfd-${var.env_config.env}-etl-app" }, var.env_config.tags)
+}
+
+resource "aws_network_acl_rule" "rda_known" {
```

Comment:
Someone help me out here pls. If we set a NACL, I think that impacts any device / host that is assigned to that subnet. Would this then block outbound 5433 / 5432 which is required for the pipeline app to reach the DB? Perhaps I'm misunderstanding NACL here.

Reply:
SG is instant level, NACL is subnet level.

Reply:
Correct, this should block all communication to hosts on the given subnet except over port 443.  I choose a NACL because we can specify DENY rules with that, from my understanding we can only define ALLOW rules with a SG.

The subnet I'm planning on setting in the `mpm_rda_cidr_block` should be limited to only hosts available via the VPC peering connection to the MPSM environment.  Are you thinking there will be an overlap?